### DISCUSSION TITLE
Pagination: onPageSizeChange required but not needed with hidePageSizeSelect

### DISCUSSION THREAD

**Discussion by edoardo** [dhis2/notes#295] [Q&A]
I'm using the `Pagination` widget with the `hidePageSizeSelect`, but I'm getting a warning about `onPageSizeChange` being required.
Currently to silence the warning I'm doing this:
>     <Pagination
>          page={page}
>          pageCount={pager.pageCount}
>          pageSize={pager.pageSize}
>          total={pager.total}
>          onPageChange={onPageChange}
>          onPageSizeChange={Function.prototype}                                                                                                                                                                  
>          hidePageSizeSelect
>          pageSummaryText={interpolationObject =>
>              i18n.t(
>                  '{{firstItem}}-{{lastItem}} of {{total}}',
>                  interpolationObject
>              )
>          }
>      />

Should `onPageSizeChange` **not** be a required prop?

**Comment by varl** [comment]
Will check, but I think it can be optional.

**Comment by khalilurrrahmanridoykhan** [comment]
Yes—`onPageSizeChange` should be optional when the page-size control is hidden, and that is how the current DHIS2 UI component behaves.

The current implementation renders `PageSizeSelect` only when `hidePageSizeSelect` is false ([source](https://github.com/dhis2/ui/blob/fc2740b54a672583df936d1f53b25a1620aafde9/components/pagination/src/pagination.js#L60-L72)). Its PropTypes declaration now defines `onPageSizeChange` as optional ([source](https://github.com/dhis2/ui/blob/fc2740b54a672583df936d1f53b25a1620aafde9/components/pagination/src/pagination.js#L129-L150)), and the TypeScript definition also uses `onPageSizeChange?`.

Therefore, with a current `@dhis2/ui` release, this is sufficient:

```jsx
<Pagination
    page={page}
    pageCount={pager.pageCount}
    pageSize={pager.pageSize}
    total={pager.total}
    onPageChange={onPageChange}
    hidePageSizeSelect
/>
```

No `Function.prototype` placeholder is needed. If the warning still appears, check for an older or duplicated `@dhis2/ui` package in the dependency tree and upgrade/deduplicate it; the warning would be coming from that older component version.
